backend/src/controllers/regionController.js

```diff
@@ -60,208 +150,124 @@
                     FROM Subregions
                     WHERE simplified_geom IS NOT NULL;
                 `;
-                const result = await sequelize.query(query, {
-                    replacements: {regionId: regionId},
-                    type: QueryTypes.SELECT
-                });
-                const result_geometry = result[0].geometry;
-                // Update the geometry of the region
-                if (result_geometry) {
-                    geometries.push(result_geometry);
-                    // Asynchronously update the geometry of the region
-                    Region.update({geom: result_geometry}, {
-                        where: {id: regionId}
-                    }).then().catch(err => console.log(err));
-                }
-            }
-        }
+      const result = await sequelize.query(query, {
+        replacements: { divisionId },
+        type: QueryTypes.SELECT,
+      });
+      const resultGeometry = result[0].geometry;
+      // Update the geometry of the region
+      if (resultGeometry) {
+        geometries.push(resultGeometry);
+        // Asynchronously update the geometry of the region
+        Region.update({ geom: resultGeometry }, {
+          where: { id: divisionId },
+        }).then().catch((err) => console.log(err));
+      }
     }
+    return true;
+  });
 
-    geometries = geometries.filter(g => g != null);
+  const results = await Promise.all(promises);
+  const allHasGeometry = results.every((r) => r !== false);
```

Comment:
**Suggestion:** Use a more descriptive variable name than 'r' for better readability.
```suggestion
  const results = await Promise.all(promises);
  const allHasGeometry = results.every((result) => result !== false);
```